backend/logic/discount/index.js

```diff
@@ -195,25 +172,25 @@ const validateDiscountOnOrder = async (
 
   const donation = get(cart, 'donation', 0)
 
-  const calculatedTotal = subTotal + shipping - discount + donation + totalTaxes
+  // Note: By calling Math.max, we don't let the amount go negative in case a
+  // discount larger than the total is applied.
+  const calculatedTotal = Math.max(
+    0,
+    subTotal + shipping - discount + donation + totalTaxes
```

Comment:
Know this was touched upon in discord this morning, but do we really want to apply discount after shipping?  Shipping is generally a fixed cost which could end up costing the seller.  

If it's fine with everyone, it's fine with me, just through I'd confirm.

Reply:
I created this separate issue to track this: https://github.com/OriginProtocol/dshop/issues/745